python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -434,34 +304,28 @@ def predict(self, X, predict_model="GPU", algo='auto',
         y : Dask cuDF dataframe  or CuPy backed Dask Array (n_rows, 1)
         """
         if predict_model == "CPU":
-            preds = self._predict_using_cpu(X, convert_dtype=convert_dtype)
+            preds = self.predict_model_on_cpu(X, convert_dtype=convert_dtype)
 
         else:
             preds = \
-                self._predict_using_fil(X, predict_model=predict_model,
-                                        algo=algo,
-                                        convert_dtype=convert_dtype,
-                                        fil_sparse_format=fil_sparse_format,
-                                        delayed=delayed)
+                self.predict_using_fil(X, predict_model=predict_model,
+                                       algo=algo,
+                                       convert_dtype=convert_dtype,
+                                       fil_sparse_format=fil_sparse_format,
+                                       delayed=delayed)
         return preds
 
-    def _predict_using_fil(self, X, predict_model="GPU", algo='auto',
-                           convert_dtype=True, fil_sparse_format='auto',
-                           delayed=True):
-        self._concat_treelite_models()
-        data = DistributedDataHandler.create(X, client=self.client)
-        self.datatype = data.datatype
-
-        kwargs = {"convert_dtype": convert_dtype,
-                  "predict_model": predict_model, "algo": algo,
-                  "fil_sparse_format": fil_sparse_format}
-        return self._predict(X, delayed=delayed, **kwargs)
+    def predict_using_fil(self, X, delayed, **kwargs):
+        self.local_model = self._concat_treelite_models()
```

Comment:
Should this cache the model and reuse local_model if one already exists? Would be nice to be able to reuse it in multiple predict calls.